osu.Game/Online/Metadata/OnlineMetadataClient.cs

```diff
@@ -116,6 +167,71 @@ public override Task<BeatmapUpdates> GetChangesSince(int queueId)
             return connection.InvokeAsync<BeatmapUpdates>(nameof(IMetadataServer.GetChangesSince), queueId);
         }
 
+        public override Task UpdateActivity(UserActivity? activity)
+        {
+            if (connector?.IsConnected.Value != true)
+                return Task.FromCanceled(new CancellationToken(true));
+
+            Debug.Assert(connection != null);
+            return connection.InvokeAsync(nameof(IMetadataServer.UpdateActivity), activity);
+        }
+
+        public override Task UpdateStatus(UserStatus? status)
+        {
+            if (connector?.IsConnected.Value != true)
+                return Task.FromCanceled(new CancellationToken(true));
+
+            Debug.Assert(connection != null);
+            return connection.InvokeAsync(nameof(IMetadataServer.UpdateStatus), status);
+        }
+
+        public override Task UserPresenceUpdated(int userId, UserPresence? presence)
```

Comment:
I think this is fine to start with, but we may want to batch this in the future depending on the overhead of individual messages in SignalR. Maybe it's good enough for this kinda throughput though, we'll see!